src/riak_cs_mp_utils.erl

```diff
@@ -302,21 +304,21 @@ write_new_manifest(M, Opts, RcPidUnW) ->
             try
                 Acl = case proplists:get_value(acl, Opts) of
                           undefined ->
-                              % 4th arg, pid(), unused but honor the contract
                               riak_cs_acl_utils:canned_acl("private", Owner, undefined);
                           AnAcl ->
                               AnAcl
                       end,
-                ClusterId = riak_cs_config:cluster_id(?PID(RcPid)),
-                M2 = M?MANIFEST{acl = Acl,
-                                cluster_id = ClusterId,
-                                write_start_time=os:timestamp()},
-                {Bucket, Key} = M?MANIFEST.bkey,
+                BagId = riak_cs_mb_helper:choose_bag_id(block, {Bucket, Key, UUID}),
+                M2 = riak_cs_lfs_utils:set_bag_id(BagId, M),
+                ClusterId = riak_cs_mb_helper:cluster_id(BagId),
+                M3 = M2?MANIFEST{acl = Acl,
+                                 cluster_id=ClusterId,
+                                 write_start_time=os:timestamp()},
```

Comment:
I understood the bug. It should be fixed by the commit https://github.com/basho/riak_cs_multibag/commit/fe84abc11330c8741825832b4bf470a50d589893 . Thanks for finding it :sunflower: